Fix invalid project state group rejects

ja00069-Git opened the pull request:
Implemented the functionality to redirect the group manager who is rejecting to be  on the the project to select a new lead before they can reject the invitation. This makes prevents the project invalid state.

eolutoyin committed:
Implemented prevention of deleting a group when the manager of the group is the lead for a project and fixed bug when updating group unchecked members in the selected members area are not removed from the group

ja00069-Git committed:
Addressed the peoject invalid state when groups rejects to be on the project and it's manager is the project lead.

1)  Implemented the functionality to redirect the group manager who is rejecting to be  on the the project to select a new lead before they can reject the invitation. This makes prevents the project invalid state.

2) Fixed compiler error caused by adding new fields to controllers that wasn't addressed in the tests classes for Gorup controller, and Home controller.

jacobevans1 reviewed (changes requested):
Approval rejection process looks solid, however a few changes are needed. 

1. A user who is not listed as the project lead should not be able to edit a project. This also creates a problem when creating a project and setting a project lead that is not the current user. 

2. When Creating/Editing a project clicking the save button does not add the groups to the project if they are not checked. They should be added if they are on the assigned side even if they are not checked. This was just fixed in the group page if you need a reference.

jacobevans1 committed:
Removed permission for a manager on a project to edit

1) Removed permission for a manager on a project who is not the lead to edit the project

ja00069-Git committed:
Groups On the assigned groups list should submit with the project.

1)  When adding a projet groups on the selected groups list submits with project even thought they are not checked

ja00069-Git committed:
Unckecked group on assigned group list now also submits with the form

1) When editing the project the groups on the assigned group list sunmits with the form even if they are ckecked or unckecked when submiting.

jacobevans1 reviewed (approved):
Changes successfully implemented and work as expected.
